src/platforms/common/server/kms-utils/kms_connector.cpp

```diff
@@ -109,6 +109,19 @@ std::tuple<mgk::DRMModeCrtcUPtr, int> find_crtc_and_index_for_connector(
 
     auto const encoders = connector_available_encoders(resources, connector.get());
 
+    // This workaround feels like overkill, but I've not fould a better place to implement a fix...
+    //
+    // We don't "seize" crtcs when we find them which means they remain "available". So there is
+    // an opportunity for the client logic to request several crtcs before using the first makes it
+    // "unavailable" and get the same crtc for multiple requests.
+    //
+    // By ensuring we don't return the same crtc until we've exhausted all alternatives we prevent
+    // the client logic using it twice.
+    //
+    // Using "thread_local" as each group of outputs is accessed on its own thread.
+    thread_local std::list<u_int32_t> recent_crtc_ids;
+    std::list<std::tuple<mgk::DRMModeCrtcUPtr, int>> skipped_crtcs;
```

Comment:
Why isn't `skipped_crtcs` also `thread_local`?

Reply:
Because it doesn't have static duration